oauth.go

```diff
@@ -0,0 +1,41 @@
+package pluginapi
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+)
+
+// UserService exposes methods to manipulate OAuth Apps.
+type OAuthService struct {
+	api plugin.API
+}
+
+// Create creates a new OAuth App,
```

Comment:
```suggestion
// Create creates a new OAuth App.
```

Reply:
Do we ever need to patch an existing OAuth App?

Reply:
Maybe. Let me just add a method for that to be sure.

Reply:
Added an `Update` method.